base/arithmetic.h

```diff
@@ -0,0 +1,146 @@
+#ifndef SLINKY_BASE_ARITHMETIC_H
+#define SLINKY_BASE_ARITHMETIC_H
+
+#include <cmath>
+#include <cstddef>
+#include <limits>
+
+namespace slinky {
+
+// Signed integer division in C/C++ is terrible. These implementations
```

Comment:
nit: it might be worth adding more meat to the bone about why it's terrible (ie what makes it unsuitable for slinky)